### ISSUE TITLE
Questions about training objective

### ISSUE THREAD

**Issue by why986** [deepbrainai-research/float#19]
<p align="center">
  <img src="https://github.com/user-attachments/assets/28d581dd-9c8a-4800-bdcf-64502c244351" alt="Training Objectives" width='50%' height='50%'/>
</p>
Hello authors! I sincerely appreciate your great work. 

I have a question about the training objective in the paper, as illustrated in the figure above. When calculating $L_{OT}$ and $L_{vel}$, do $v_t$ and $u_t$ contain the preceding motion? In other words, their shape is $[(L'+L), d]$ or $[L, d]$? 

Additionally, I am also confused in the motion vector $w_{r\rightarrow D^{1:L}}$. Is $w_{r\rightarrow D^{1:L}}$ predicted frame by frame? $w_{r\rightarrow D^{i}}$ is predicted by first feeding $D^i$ to motion_autoencoder.enc.enc_motion() and then feeding the encoded $\lambda_m(D^i)$ to motion_autoencoder.dec.direction(). Did you use `h_start` when predicting $w_{r\rightarrow D^{1:L}}$?

**Comment by Egoago** [comment]
Check out #4

**Comment by why986** [comment]
Hi @Egoago ,

Thank you for your reply. According to #4 , I acknowledge that the preceding motion $w_{r\rightarrow D^{-L':0}}$ is clean during training. Following this, the preceding motion is not included when calculating $L_{OT}$. 

However, I am still confused about the generation of the motion vector $w_{r\rightarrow D^{1:L}}$. I wonder if you already implemented the training algorithm of FLOAT?

**Comment by Egoago** [comment]
The author pointed out that there's a discrepancy between the published code, the text and also the images of the paper. While we are waiting for the revision of the paper, I implemented my own version. Sadly Im not allowed to share it with you due to company policy, hope you understand.

**Comment by TaekyungKi** [comment]
@Egoago @why986 
Hi, Sorry for the late update. The revision is uploaded in arXiv https://arxiv.org/abs/2412.01064.
Hope this helps!
If you have any further questions, please let me know.

**Comment by why986** [comment]
Thank you @TaekyungKi . I will read the updated version carefully! Congratulations to you for the acceptance by ICCV 2025!

**Comment by why986** [comment]
Thank you @TaekyungKi , your revision have resolved my questions! However, I am interested in your training settings. Did you use any learning rate scheduler (e.g., CosineAnnealingLR) during the training phase?

**Comment by TaekyungKi** [comment]
@why986 Thank you!
I hadn’t tried any learning rate schedulers — constant learning has been sufficient for our case.
I’m not 100% sure, but it might help even more.

**Comment by why986** [comment]
@TaekyungKi Thank you for your reply! I am impressed by FLOAT's superior performance, and therefore, I try to train FLOAT from scratch for further research. But my checkpoints struggle to produce clear results, and there are always blurred artifacts, especially inside the mouths. I would greatly appreciate it if you could offer me some suggestions!

**Comment by TaekyungKi** [comment]
@why986 
Ok,
As the current version of the motion latent AE is trained only on open-sourced datasets, which may lack highly detailed inner mouth data, one straightforward solution is to further train it with a larger, high-quality dataset (But yes I know this is a costly solution). I recommend using a dataset like MEAD for its extremely clear inner mouth visuals (while being cautious about overfitting at the same time).
In my case, L_{comp-lp} is very critical for improving lip and mouth fidelity. FYI, It has little effect during the early training phase, but it suddenly improves mouth and teeth fidelity—possibly after 180k iterations.

**Comment by why986** [comment]
@TaekyungKi Thank you! I appreciate your useful advice!
